controllers/spacebindingrequestmigration/spacebindingrequest_migration_controller.go

```diff
@@ -0,0 +1,182 @@
+			return reconcile.Result{}, nil
+		}
+		// error while reading MUR
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound MUR")
+	}
+
+	// skip workspace creator spacebinding
+	// the controller will convert only spacebindings created by system admins using the sandbox-cli.
+	// If the creator label on the space matches the usersignup reference on the MUR then this is the owner of the space
+	// and the spacebinding should not be migrated.
+	if mur.ObjectMeta.OwnerReferences == nil || len(mur.ObjectMeta.OwnerReferences) == 0 {
+		return ctrl.Result{}, errs.New("MasterUserRecord has no UserSignup owner reference")
+	}
+	// skip spaces with no creator label
+	if _, ok := space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey]; !ok {
+		// let's log an error, since this should not happen in production
+		logger.Error(errs.New("space has no SpaceCreatorLabelKey set"), "the spacebindings for this space will not be migrated", "space name", space.Name)
+		return ctrl.Result{}, nil
+	}
+	usersignup := mur.ObjectMeta.OwnerReferences[0]
+	if space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey] == usersignup.Name {
+		return reconcile.Result{}, nil
+	}
+
+	// get the spaceRole
+	spaceRole := spaceBinding.Spec.SpaceRole
+
+	// get member cluster name where the space was provisioned
+	targetCluster := space.Spec.TargetCluster
+	memberCluster, memberClusterFound := r.MemberClusters[targetCluster]
+	if !memberClusterFound {
+		return ctrl.Result{}, errs.New(fmt.Sprintf("unable to find member cluster: %s", targetCluster))
+	}
+
+	// get the home namespace from space
+	defaultNamespace := getDefaultNamespace(space.Status.ProvisionedNamespaces)
+
+	// construct a SpaceBindingRequest object
+	sbrName := mur.GetName() + "-" + spaceRole
+	sbr := &toolchainv1alpha1.SpaceBindingRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sbrName,
+			Namespace: defaultNamespace,
+		},
+		Spec: toolchainv1alpha1.SpaceBindingRequestSpec{
+			MasterUserRecord: mur.GetName(),
+			SpaceRole:        spaceRole,
+		},
+	}
+
+	// check if SpaceBindingRequest doesn't exist already
+	err = memberCluster.Client.Get(ctx, types.NamespacedName{
+		Namespace: defaultNamespace,
+		Name:      sbrName,
+	}, &toolchainv1alpha1.SpaceBindingRequest{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// create the sbr resource
+			if err := memberCluster.Client.Create(ctx, sbr); err != nil {
+				return ctrl.Result{}, errs.Wrapf(err, "unable to create SpaceBindingRequest")
+			}
+
+			// let's requeue after we created the SBR, so that in next loop the migrated SpaceBinding object will be deleted
+			return ctrl.Result{Requeue: true}, nil
+		}
+		// Error reading the object
+		return ctrl.Result{}, err
+	}
+
+	// if the SBR was found ( was created from the previous reconcile loop), we can now delete the SpaceBinding object
+	if err := r.Client.Delete(ctx, spaceBinding); err != nil {
+		return ctrl.Result{}, errs.Wrapf(err, "unable to delete the SpaceBinding")
+	}
```

Comment:
if `err` here is `NotFound`, should we be ok with it?

```suggestion
	if err := r.Client.Delete(ctx, spaceBinding); err != nil {
	        if !errors.IsNotFound(err) {
			return ctrl.Result{}, errs.Wrapf(err, "unable to delete the SpaceBinding")
		}
	}
```

Reply:
good catch ! I've applied your first suggestion. please see https://github.com/codeready-toolchain/host-operator/pull/922/commits/c94803b6e8bad98a6f126c635d98ae601079ff8a

thanks!